docs/1-web-api-server.md

```diff
@@ -0,0 +1,102 @@
+# Web API server
+
+### Append (to) keymap
+
+This endpoint is responsible for "Add a value `A` for key `X`, where key Y has value `B`". 
+If the key-binding does not exist, a new keymap is created with the `Y` key and the `B` value.
+
+* HTTP method: **POST**
+* Method URL: `/`
+* Accepted content types: `application/json`
+
+#### Append is supported for:
+
+* append - used to add single key-binding to existing keymap
+* add - used to add new keymap
+
+#### Request Body: 
+
+* where (required)
+    * must contain one pair key-value e.g. `Y`: `B`
+* append (optional)
+    * must contain one pair key-value for append e.g. `X`: `A`
+
+```
+{
+    "where": {
+        "Y": "B"
+    },
+    "append": {
+         "X": "A"
+    }
+}
+```
+
+#### Response:
+```
+{
+    "result": {
+        "X": "A"
+    },
+    "error": null
+}
+```
+
+When no include is specified, a new keymap is created.
+```
+{
+    "where": {
```

Comment:
Shouldn't this be `"append"` instead of `"where"`?

Reply:
The order of the "where" and "append" parameters is arbitrary. I wrote "where" at the top because "where" is required.